lib/utils.pm

```diff
@@ -3453,4 +3458,100 @@ sub upload_folders {
     }
 }
 
+sub _flush_console {
+    my $buf;
+    my $ret = '';
+
+    while ($buf = wait_serial(qr/.+/s, timeout => 1, quiet => 1, record_output => 1)) {
+        $ret .= $buf;
+    }
+
+    return $ret;
+}
+
+sub _cmd_run_impl {
+    my ($cmd, %args) = @_;
+
+    $args{timeout} //= $bmwqemu::default_timeout;
+    die "Terminator '&' found in cmd_run call. cmd_run can not check script success. Use 'background_script_run' instead."
+      if $cmd =~ m/(?<!\\)&\s*$/;
+
+    if (is_serial_terminal()) {
+        wait_serial(serial_terminal::serial_term_prompt(), no_regex => 1, quiet => 1, timeout => 5) or die 'Terminal not ready';
+    }
+
+    my $marker = hashed_string("CR" . $cmd . $args{timeout});
```

Comment:
Is this approach collision safe?  I'd like to be able to use this function both as root & rootless in container tests.  I'm trying to have this fixed in script_run using a monotonically increasing marker:

https://github.com/os-autoinst/os-autoinst/pull/2734

Reply:
@ricardobranco777 ping for the failed job, please.